### ISSUE TITLE
ansible Python API does not return completely

### ISSUE THREAD

**Issue by academciaingao** [ansible/ansible#36942]
##### ISSUE TYPE

 - Bug Report

##### ANSIBLE VERSION

```
2.4
```

##### STEPS TO REPRODUCE
we build a app auto-deploying system base on ansible python api refers to the example on "http://docs.ansible.com/ansible/latest/dev_guide/developing_api.html#python-api-2-0"

the core feature of our system is using ansible python api to run playbooks and get results in json.

for most situation, it works well, but recently, we got a problem.

one playbook has a part using do-until loops, like this:

- name: check file
  shell: ls -l | grep /tmp/x.do | wc -l
  register: result
  until: result.stdout == "1"
  retries: 10
  delay: 2

this task is to check another task's status, the if the task succeeds, file /tmp/x.do will be created. 

this playbook works well if i use command "ansible-play ......." in os shell, the "check file" task continuously run shell command until file /tmp/abc created. The finally result  of this playbook is OK. 

##### EXPECTED RESULTS
ansible python api return the final result after the do until loop complete

##### ACTUAL RESULTS
 if run the playbook using python api, the return is not complete. The do-until loop executes only once and return directly. The result shows task "check file" failed, because in first check, the file has not created. The finally result  of this playbook is failed.

**Comment by ansibot** [comment]
@academciaingao Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
List Information
================

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a developer question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.

   * IRC: #ansible-devel on irc.freenode.net   
   * IRC: #ansible on irc.freenode.net   
   * mailing list: https://groups.google.com/forum/#!forum/ansible-devel
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!
